docs/refs.bib

```diff
@@ -571,6 +571,14 @@ @misc{Hall_2006_Indecomposable
   primaryClass={quant-ph}
 }
 
+@article{Horn_1985_Matrix,
+  title = {Matrix Analysis},
+  author = {Horn, Johnson},
+  year = {1985},
+  publisher = {Cambridge University Press},
+  doi = {https://doi.org/10.1017/CBO9780511810817},
```

Comment:
Generally, DOIs are keyed in not as URLs but as the identifiers themselves. So for instance 

```suggestion
  doi = {10.1017/CBO9780511810817},
```

This is also true for the other references you have added as well.